pkg/arvo/ted/eth-watcher.hoon

```diff
@@ -13,7 +13,7 @@
 =/  m  (strand:strandio ,vase)
 ^-  form:m
 ;<  =latest=block  bind:m  (get-latest-block:ethio url.pup)
-;<  pup=watchpup   bind:m  (zoom pup number.id.latest-block)
+;<  pup=watchpup   bind:m  (zoom pup ?~(to.pup number.id.latest-block u.to.pup))
```

Comment:
Not sure this is quite right. Two things:
- Since we cannot look at blocks beyond the latest existing block, I think you want this to be `=+(last=number.id.latest-block (min last (fall to.pup last))`.
- `+zoom` notably _approaches_ the given block, but doesn't actually reach it. This means that, as currently written, we will only get logs from up to `(sub to.pup zoom-margin)`.

Might be cleanest to lightly rewrite `+zoom` to take an additional `up-to=number:block` argument, including that in its `up-to-block` calculation, to ensure that it stays the margin away from the network's latest block, but does eventually reach up to exactly the block we specify.

Reply:
Also wondering if, once we fetched the `to.pup` block, we should signal "completion" of this watchpup somehow. Right now it'll probably stick around and continue checking the latest block number, but once it's out of range, it'll be out of range forever. Not a super heavy load, but would be nice to clean up if we can.

Reply:
Fixed, and decided to check for completion on behn `%wake`, where we kill and decide not to restart the thread if the pup is complete.